pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -536,7 +524,7 @@ type EC2SDConfig struct {
 	// https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeInstances.html
 	// Filter API documentation: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_Filter.html
 	// +optional
-	Filters []*EC2Filter `json:"filters"`
+	Filters []Filter `json:"filters,omitempty"`
```

Comment:
It's not enforced by Prometheus (e.g. you can define several times the same fliter name in the list) but IMHO it makes sense to add some restriction at the API level here.
```suggestion
	// +listType:=map
	// +listMapKey:=Name
	Filters []Filter `json:"filters,omitempty"`
```

Reply:
Seems like `listMapKey` can only be applied for associative lists 🤔

Reply:
Getting this error: `must apply listMapKey to an associative-list`